src/utilities/include/public.h

```diff
@@ -166,36 +182,47 @@ struct Mesh {
   /// as 3 * tri + i, representing the tangent from Mesh.triVerts[tri][i] along
   /// the CCW edge. If empty, mesh is faceted.
   std::vector<glm::vec4> halfedgeTangent;
-};
 
-/**
- * An alternative to Mesh for output suitable for pushing into graphics
- * libraries directly.
- */
-class MeshGL {
- public:
-  MeshGL(int _numVert, int _numTri)
-      : numVert(_numVert),
-        numTri(_numTri),
-        vertPos_(std::make_unique<float[]>(3 * _numVert)),
-        vertNormal_(std::make_unique<float[]>(3 * _numVert)),
-        triVerts_(std::make_unique<uint32_t[]>(3 * _numTri)) {}
-
-  /// A flat buffer of positions, length 3 * numVert [x, y, z, x, y, z, ...].
-  float* vertPos() const { return this->vertPos_.get(); }
-  /// A flat buffer of normals, length 3 * numVert [x, y, z, x, y, z, ...].
-  float* vertNormal() const { return this->vertNormal_.get(); }
-  /// A flat buffer of verts, length 3 * numTri [0, 1, 2, 0, 1, 2, ...].
-  uint32_t* triVerts() const { return this->triVerts_.get(); }
-  /// Number of vertices
-  const int numVert;
-  /// Number of triangles
-  const int numTri;
+  Mesh() = default;
+  Mesh(const std::vector<glm::vec3>& vertPos_,
+       const std::vector<glm::ivec3>& triVerts_,
+       const std::vector<glm::vec3>& vertNormal_ = {},
+       const std::vector<glm::vec4>& halfedgeTangent_ = {})
+      : vertPos(vertPos_),
+        triVerts(triVerts_),
+        vertNormal(vertNormal_),
+        halfedgeTangent(halfedgeTangent_) {}
+
+  Mesh(const MeshGL& in) {
```

Comment:
Okay, I lied, this is another extraneous copy. Hopefully doesn't hurt perf too much, but we can always come back and optimize it out later.